torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -116,12 +116,15 @@ def set_shuffle(self, shuffle=True):
 
     def set_seed(self, seed: int):
         self._seed = seed
+        return self
 
     def __iter__(self) -> Iterator[T_co]:
         if not self._enabled:
             for x in self.datapipe:
                 yield x
         else:
+            if self._seed is None:
+                self._seed = int(torch.empty((), dtype=torch.int64).random_().item())
```

Comment:
1. If I understand this implementation correctly - even if the user doesn't set seed for every iteration, a new seed will still be created and used.
2. I actually think the behavior may not be different from the existing implementation where a new seed is created during `reset()`, which I believe is triggered lazily as well. Not 100% confident though.
3. This is in contrast with the other PR that keeps the seed fix unless the `set_seed` is called. Any reason why you changed your mind?

Reply:
I am changing it because `reset` is not invoked during the state of `NotStarted`. (It was a bug and fixed by this PR)

> This is in contrast with the other PR that keeps the seed fix unless the set_seed is called. Any reason why you changed your mind?

The other PR keeps seed fix unless the set_seed is called because that's the only way to align the behavior between `MapDataPipe.shuffle` and `IterDataPipe.shuffle`, when `MapDataPipe.shuffle` is still a `MapDataPipe`. Let's talk about it in the meeting.

Reply:
Having different seeds is preferred. But, the other PR is a compromisation for the sake of behavioral alignment.